src/main/java/Duke.java

```diff
@@ -1,10 +1,191 @@
+        return userInput;
+    }
+
+    private static void printWelcomeMsg() {
+        System.out.println("Hello! I am Kaman :)");
+        System.out.println("What can I do for you?");
+    }
+
+    private static void markAsDone(int index) {
+        if (index < 0 || index >= records.size()) {
+            System.out.println("Invalid input! (Index cannot be out of bounds)");
+            return;
+        }
+        records.get(index).setAsDone();
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println("\t" + records.get(index));
+    }
+
+    private static void addRecord(String[] detailFragments, String taskType) {
+        String[] details = null;
+        String taskName;
+        String date;
+        switch (taskType) {
+        case Todo.TASK_TYPE:
+            taskName = String.join(" ", detailFragments);
+            records.add(new Todo(taskName));
+            break;
+        case Deadline.TASK_TYPE:
+            details = getTaskNameAndDate(detailFragments, taskType);
+            if (details != null) {
+                taskName = details[0];
+                date = details[1];
+                records.add(new Deadline(taskName, date));
+            }
+            break;
+        case Event.TASK_TYPE:
+            details = getTaskNameAndDate(detailFragments, taskType);
+            if (details != null) {
+                taskName = details[0];
+                date = details[1];
+                records.add(new Event(taskName, date));
+            }
+            break;
+        }
+        if (details != null || taskType.equals(Todo.TASK_TYPE)) {
+            int numberOfRecords = records.size();
+            System.out.println("Got it. I've added this task:");
+            System.out.println("\t" + records.get(numberOfRecords - 1));
+            System.out.printf("Now you have %d tasks in the list.\n", numberOfRecords);
+        }
+    }
+
+    private static String[] getTaskNameAndDate(String[] detailFragments, String taskType) {
+        int slashElementIndex;
+        boolean hasSlashElement = false;
+        boolean isTypeMatch = false;
+        for (slashElementIndex = 0; slashElementIndex < detailFragments.length - 1; slashElementIndex++) {
+            if (detailFragments[slashElementIndex].charAt(0) == '/') {
+                hasSlashElement = true;
+                break;
+            }
+        }
+        if (!hasSlashElement) {
+            System.out.println("Invalid input! (No date/time provided)");
+            return null;
+        }
+
+        switch (taskType) {
+        case Deadline.TASK_TYPE:
+            if (detailFragments[slashElementIndex].substring(1).equals("by")) {
+                isTypeMatch = true;
+            }
+            break;
+        case Event.TASK_TYPE:
+            if (detailFragments[slashElementIndex].substring(1).equals("at")) {
+                isTypeMatch = true;
+            }
+            break;
```

Comment:
I believe a "default" case should be added here, so as to adhere to the coding standard on using a switch statement.